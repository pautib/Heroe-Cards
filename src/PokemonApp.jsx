import {AppRouter} from "./router/AppRouter.jsx";
import {AuthProvider} from "./auth/index.js";

export const HeroesApp = () => {
    return (
        <AuthProvider>
            <AppRouter/>
        </AuthProvider>
    )
}