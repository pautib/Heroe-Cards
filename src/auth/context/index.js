export * from './AuthContext.jsx';
export * from './AuthProvider.jsx';
export * from './authReducer.js';