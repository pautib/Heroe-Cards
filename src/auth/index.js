export * from './pages'
export * from './context'
export * from './types/types.js'
//export * from './helpers'
//export * from './components'
//export * from './hooks'