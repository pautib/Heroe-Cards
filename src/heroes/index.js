export * from './pages'
export * from './routes'
//export { heroes } from './data/heroes.js'
//export * from './helpers'
//export * from './components'
//export * from './hooks'