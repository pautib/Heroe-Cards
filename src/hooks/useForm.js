import {useState} from "react";

export const useForm = (initialForm = {}) => {

    const [formState, setFormState] = useState(initialForm);

    const onInputChange = ({ target }) => {
        //console.log(target)
        const {name, value} = target;
        setFormState({...formState, [ name ]: value});
    }

    const onResetForm = () => {
        setFormState(initialForm);
    }

    return {
        ...formState,
        formState,
        onInputChange,
        onResetForm,
    }
}