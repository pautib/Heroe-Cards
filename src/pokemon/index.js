export * from './pages'
export * from './routes'
//export * from './helpers'
//export * from './components'
//export * from './hooks'