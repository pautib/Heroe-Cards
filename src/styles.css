@font-face {
    font-family: 'Pokemon';
    src: url('/PokemonSolid.ttf'); /* Safari, Android, iOS */
}

.spinner {
    animation: spin 1.2s linear infinite;
    position: absolute;
    top: 40%;
    left: 40%;
}

@keyframes spin {

    100% {
        transform: rotate(360deg);
    }
}

.pokeNavSection {
    font-size: 30px;
    font-family: 'Pokemon';
    color: #f2c304 !important;
    -webkit-text-stroke: 2px #3c5aa6;
}


.pokeNavSection.active {
    text-decoration: underline;
}

.pokeNavSection:hover {
    font-size: 31px;
}

.pokeNatureButtons {
    border: thick solid #3c5aa6;
    color: #f2c304 !important;
    font-weight: bold;
}

.pokeNatureButtons:focus {
    background-color: #aa0d18;
}

.pokeGoBackButton {
    border: thick solid #3c5aa6;
    color: #f2c304 !important;
    font-weight: bold;
}

.pokeGoBackButton:hover {
    background-color: #aa0d18;
}

.pokeNavWalletButton {
    border: thick solid black;
    color: #f2c304 !important;
    background-color: #aa0d18;
}

.pokeNavWalletButton:hover {
    background-color:  #3c5aa6;
    width: 101%;
}

.pokeNavWalletButton.active {
    background-color:  #3c5aa6;
}


