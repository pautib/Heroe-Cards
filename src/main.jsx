import React from 'react'
import ReactDOM from 'react-dom/client'
import './styles.css'
import {HeroesApp} from "./HeroesApp.jsx";
import {BrowserRouter} from "react-router-dom";

ReactDOM.createRoot(document.getElementById('root')).render(
  //<React.StrictMode>
    <BrowserRouter>
        <HeroesApp />
    </BrowserRouter>
  //</React.StrictMode>,
)
