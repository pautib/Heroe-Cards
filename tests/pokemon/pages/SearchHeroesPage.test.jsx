import {fireEvent, render, screen} from "@testing-library/react";
import {AuthContext} from "../../../src/auth";
import React from "react";
import {MemoryRouter, useNavigate} from "react-router-dom";
import {Navbar} from "../../../src/ui";
import {SearchPokemonPage} from "../../../src/pokemon/index.js";
import {useForm} from "../../../src/hooks/useForm.js";

const mockUseNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockUseNavigate,
}))

jest.mock("../../../src/hooks/useForm.js");
describe('Pruebas en SearchHeroesPage', () => {

    beforeEach(() => jest.clearAllMocks());

    test('Debe de mostrar los valores por defecto', () => {

        const { container } = render(
            <MemoryRouter>
                <SearchPokemonPage />
            </MemoryRouter>
        );

        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar a Batman y el input con el valor del queryString', () => {

        render(
            <MemoryRouter initialEntries={['/search?query=batman']}>
                <SearchPokemonPage />
            </MemoryRouter>
        );

        expect(screen.getByText('Batman')).toBeTruthy();
        expect(screen.getByRole('img').src).toContain("/heroes/dc-batman.jpg");
        const noDisplayHeroDiv = screen.getByLabelText('nohero');
        expect(noDisplayHeroDiv.style.display).toBe('none');
    });

    test('Debe de mostrar un error si no se encuentra el heroe', () => {

        render(
            <MemoryRouter initialEntries={['/search?query=dummyHero']}>
                <SearchPokemonPage />
            </MemoryRouter>
        );

        const noDisplayHeroDiv = screen.getByLabelText('nohero');
        expect(noDisplayHeroDiv.style.display).toBe('');

    });

    test('Debe de llamar el navigate a la pantalla nueva', () => {

        useForm.mockReturnValue({
            searchText: 'dummyHero',
            onInputChange: jest.fn(),
        })

        render(
            <MemoryRouter initialEntries={['/search']}>
                <SearchPokemonPage />
            </MemoryRouter>
        );

        const searchBtn = screen.getByRole('button');
        fireEvent.click(searchBtn);

        expect(mockUseNavigate).toHaveBeenCalledWith("?query=dummyhero");

    })


});