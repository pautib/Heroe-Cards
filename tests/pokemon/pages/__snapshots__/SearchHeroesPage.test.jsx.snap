// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en SearchHeroesPage Debe de mostrar los valores por defecto 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form>
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          text="text"
          value=""
        />
        <button
          class="btn btn-outline-primary mt-1"
        >
          Search
        </button>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Result
      </h4>
      <hr />
      <div
        class="alert alert-primary animate__animated animate__fadeIn"
      >
        Search a heroe
      </div>
      <div
        aria-label="nohero"
        class="alert alert-danger animate__animated animate__fadeIn"
        style="display: none;"
      >
        No heroe with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
